AI/VCAI/Goals.cpp

```diff
@@ -1001,7 +1003,11 @@ TGoalVec Conquer::getAllPossibleSubgoals()
 							if (obj->ID.num == Obj::HERO) //enemy hero may move to other position
 								ret.push_back(sptr(Goals::VisitHero(obj->id.getNum()).sethero(h).setisAbstract(true)));
 							else //just visit that tile
-								ret.push_back(sptr(Goals::VisitTile(dest).sethero(h).setisAbstract(true)));
+								if(obj->ID.num == Obj::TOWN)
+									//if target is town, fuzzy system will use additional "estimatedReward" variable to increase priority a bit
+									ret.push_back(sptr(Goals::VisitTile(dest).sethero(h).setobjid(obj->ID.num).setisAbstract(true)));

```

Comment:
What's the point of this line? Where objid is checked?

We also have GetObj goal and eventually it should be used instead of VisitTile, just to evaluate underlying objects.